Source/player.cpp

```diff
@@ -4127,110 +4127,51 @@ void __fastcall SyncInitPlr(int pnum)
 
 void __fastcall CheckStats(int pnum)
 {
-	int v1; // esi
-	int v2; // eax
-	char v3; // cl
-	signed int v4; // esi
-	signed int v5; // edi
-	int v6; // edx
-	int v7; // ecx
-	int v8; // edx
-	int v9; // ecx
-	int v10; // edx
-	int v11; // ecx
-	int v12; // edx
-	int v13; // ecx
-	//signed int v14; // [esp+Ch] [ebp-4h]
-
-	v1 = pnum;
-	if ( (unsigned int)pnum >= MAX_PLRS )
+	if ( (DWORD)pnum >= MAX_PLRS ) {
 		TermMsg("CheckStats: illegal player %d", pnum);
-	v2 = v1;
-	v3 = plr[v1]._pClass;
-	if ( v3 )
-	{
-		if ( v3 == 1 )
-		{
-			v4 = 1;
-		}
-		else if ( v3 == 2 )
-		{
-			v4 = 2;
-		}
-		/*else
-		{
-			v4 = v14;
-		}*/
 	}
-	else
-	{
-		v4 = 0;
+
+	int c;
+	if ( plr[pnum]._pClass == PC_WARRIOR ) {
+		c = PC_WARRIOR;
+	} else if ( plr[pnum]._pClass == PC_ROGUE ) {
+		c = PC_ROGUE;
+	} else if ( plr[pnum]._pClass == PC_SORCERER ) {
+		c = PC_SORCERER;

```

Comment:
Really? The original code is quite funny. Why not just replace this with `int c = plr[pnum]._pClass`? It would do the same thing.

**Edit:** obviously, being a preservation project we do whatever the original did.

Reply:
They probably didn't use the enum values for the assignment, the values just happened to be the same (enums usually aren't explicitly assigned values).